### ISSUE TITLE
Converging problem for large pictures

### ISSUE THREAD

**Issue by mxmlnkn** [ComputationalRadiationPhysics/imresh#39]
With the createAtomCluster a atom cluster can be generated. The generated pictures smaller ~500x500 do converge fine, but for the version 3000x1500 it doesn't if the macro variable `SCALE_EXAMPLE_IMAGE` is set to 1.

*Ground Truth (`#define SCALE_EXAMPLE_IMAGE 1`)*
![atomcluster-scaled-object](https://cloud.githubusercontent.com/assets/6842824/12664822/20e81768-c633-11e5-985b-c9f2ad7c7bd0.png)

*Diffraction Intensity (`#define SCALE_EXAMPLE_IMAGE 1`)*
![atomcluster-scaled-diffractionintensity](https://cloud.githubusercontent.com/assets/6842824/12664823/20e8bff6-c633-11e5-8cb0-707c5bab581e.jpg)

*Reconstructed Image (`#define SCALE_EXAMPLE_IMAGE 1`) (aborted after 64 iterations) *
![atomcluster-scaled-reconstructed](https://cloud.githubusercontent.com/assets/6842824/12664821/20e74f04-c633-11e5-8b9a-aadd37db61c1.jpg)

But if `SCALE_EXAMPLE_IMAGE` is set to 0 it actually does converge exactly as well as the smaller versions.

*Ground Truth (`#define SCALE_EXAMPLE_IMAGE 0`)*
![atomcluster-extent-object](https://cloud.githubusercontent.com/assets/6842824/12664820/20e72934-c633-11e5-99f0-94fad43dbf89.png)

*Diffraction Intensity (`#define SCALE_EXAMPLE_IMAGE 0`)*
![atomcluster-extent-diffractionintensity](https://cloud.githubusercontent.com/assets/6842824/12664818/20e32f1e-c633-11e5-9278-a45932173598.jpg)

*Reconstruction (`#define SCALE_EXAMPLE_IMAGE 0`)*
![atomcluster-extent-reconstructed](https://cloud.githubusercontent.com/assets/6842824/12664819/20e60fae-c633-11e5-8cd6-a1bd999e1503.png)
Note the cluster is barely visible in the upper and bottom left corners.

I'm not sure why this happens. I hope I do not embarrass myself, but my guess actually is on the algorithm itself. First tests have shown, that it is a problem with the creation of the initial mask. The initial mask uses a a threshold determined by the maximum of the auto-correlation map (Fourier transform of diffraction intensity). For some reason that mask creation is dependent on the image dimensions. For the scaled atom cluster too many pixels will be masked out. The shrink wrap basically is too strong and slips off the object too reconstruct.

_Initial Mask (scaled = 1)_
![atomcluster-scaled-init-mask](https://cloud.githubusercontent.com/assets/6842824/12665802/bb0ab3b2-c63a-11e5-874e-3811b62a7c84.png)

_Initial Mask (scaled = 0)_
![atomcluster-extent-init-mask](https://cloud.githubusercontent.com/assets/6842824/12665801/baf58640-c63a-11e5-99e3-d47fbc461204.png)

Basically, after tweaking `rIntensityCutOffAutoCorel` it should also converge, but somehow this parameter should be chosen automatically, or another parameter should be used, e.g. to select the 50% top pixels to be masked out initially, where 50% is an adjustable parameter. (Note that on 100% and 0% the HIO algorithm may not advance).

The proposed method would need to make a histogram of the pixels though, which could be a quite expensive operation.

**Comment by Ferruck** [comment]
Where's the point of not setting `SCALE_EXAMPLE_IMAGE 0`? I mean, why should we scale the image and loose information?
